### PR TITLE
Completed the attribute support for servers.

### LINKED ISSUE #92: Complete attribute support for Servers, and Password is Optional
I tried to specify a custom server with a `privateKey`, but I got this error:

> Error: servers must contain id, username and password

Alas, this is more strict than the actual [servers](http://maven.apache.org/settings.html#Servers) definition.

Ideally, any legitimate server attribute should pass-through and `password` is _optional_, not _required_.

### PR DESCRIPTION
**Pull request by awhitford**
Note that `password` is optional because a `privateKey` may be used.
Resolves #92.

### CODE REVIEW

File: index.test.js [hunk 1]
```diff
@@ -63,11 +63,11 @@ afterAll(() => {
 
 test('run with all feature', () => {
 
-    process.env['INPUT_SERVERS'] =  '[{"id": "serverId", "username": "sUsername", "password": "sPassword", "configuration": {"props1": "value1"}}]';
-    process.env['INPUT_ORACLESERVERS'] =  '[{"id": "oServerId", "username": "oUsername", "password": "oPassword"}]';
-    process.env['INPUT_GITHUBSERVER'] =  true;
+    process.env['INPUT_SERVERS'] = '[{"id": "serverId", "username": "sUsername", "password": "sPassword", "configuration": {"props1": "value1"}}]';
```
Comment (slawekjaranowski) [inline index.test.js:66]: please add new properties to test

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]
- please add use case with new properties to `settings.test.js`
- please update `README` to show new properties
- please squash to one final commit

and thank you for PR

File: index.test.js (hunk 1 above)
Comment (awhitford) [reply index.test.js:66]: I added additional tests to `settings.test.js`, and updated the `README` to mention the additional `server` properties.

**Comment by awhitford** [comment]
Can't you just squash on the merge?

**Review by slawekjaranowski — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
@awhitford once again thanks you, I will prepare next version until week end

**Pull request merged**